common/hexutil/hexutil.go

```diff
@@ -64,6 +64,14 @@ func Decode(input string) ([]byte, error) {
 	if !has0xPrefix(input) {
 		return nil, ErrMissingPrefix
 	}
+	
+	// Check for invalid characters before passing to hex.DecodeString
+	for i := 2; i < len(input); i++ {
+		if !isHexCharacter(input[i]) {
+			return nil, ErrSyntax
+		}
+	}
+	
 	b, err := hex.DecodeString(input[2:])
```

Comment:
hex string validation errors will be caught here, and will be returned as `ErrSyntax` from `mapError` below.  Pre-validation is unnecessary.